### PR TITLE
Raise new error

### LINKED ISSUE #4792: Assembler should raise an error if set rep_delay or rep_time are not in the backend's list of acceptable values
### Information

- **Qiskit Terra version**:
- **Python version**:
- **Operating system**:

### What is the current behavior?
The assembler does not raise an error if outside of the backends acceptable values.

### Steps to reproduce the problem
See https://github.com/Qiskit/qiskit-terra/blob/2aecdcd7a9c412e5d6fd756e6cc3276ea285e957/qiskit/compiler/assemble.py#L288 and rep_times below.

### What is the expected behavior?
If `rep_times` or `rep_delays` is set, an error should be raised if an unacceptable value is provided.

### Suggested solutions
Raise an error if this scenario is encountered.

### PR DESCRIPTION
**Pull request by rochisha0**
### Summary
Fixes #4792 by @taalexander

### CODE REVIEW

**Comment by rochisha0** [comment]
@taalexander Can you please review this?

File: qiskit/compiler/assemble.py [hunk 1]
```diff
@@ -250,7 +250,8 @@ def _parse_pulse_args(backend, qubit_lo_freq, meas_lo_freq, qubit_lo_range,
         RunConfig: a run config, which is a standardized object that configures the qobj
             and determines the runtime environment.
     Raises:
-        SchemaValidationError: if the given meas_level is not allowed for the given `backend`.
+        SchemaValidationError: if the given meas_level, rep_time, rep_delay is not allowed 
+            for the given `backend`.
```
Comment (taalexander) [inline qiskit/compiler/assemble.py:254]:
```suggestion
        SchemaValidationError: if the given ``meas_level``, ``rep_time`` or ``rep_delay`` is not allowed 
            for the given ``backend``.
```

File: qiskit/compiler/assemble.py [hunk 2]
```diff
@@ -265,6 +266,18 @@ def _parse_pulse_args(backend, qubit_lo_freq, meas_lo_freq, qubit_lo_range,
                  ).format(meas_level, backend_config.backend_name, backend_config.meas_levels)
             )
 
+        if rep_time not in getattr(backend_config, 'rep_times', None):
```
Comment (taalexander) [inline qiskit/compiler/assemble.py:269]:
```suggestion
        if rep_time not in getattr(backend_config, 'rep_times', []):
```

File: qiskit/compiler/assemble.py [hunk 3]
```diff
@@ -265,6 +266,18 @@ def _parse_pulse_args(backend, qubit_lo_freq, meas_lo_freq, qubit_lo_range,
...
+            raise SchemaValidationError(
+                ('rep_time = {} not supported for backend {}, only {} is supported'
+                 ).format(rep_time, backend_config.backend_name, backend_config.rep_times)
```
Comment (taalexander) [inline qiskit/compiler/assemble.py:272]:
```suggestion
                'rep_time = {} not supported for backend {}, '
                'only {} is supported'.format(
                      rep_time, 
                      backend_config.backend_name, 
                      backend_config.rep_times,
                 )
```

Comment (taalexander) [reply qiskit/compiler/assemble.py:272]: Same below please.

**Comment by rochisha0** [comment]
@taalexander I have made the changes but still, the tests are failing.

**Comment by taalexander** [comment]
See here https://dev.azure.com/qiskit-ci/qiskit-terra/_build/results?buildId=16878&view=logs&j=8eacdd59-c2e8-5617-75a5-8ed7c854a78f&t=5df005f4-d5f4-5d2d-0307-5b24b15488fc&l=5598, it seems somehow the value `rep_time=None` is happening. rep_time is not required to be set if rep_delay is set.

**Pull request closed without merging**